src/FSharpVSPowerTools.Logic/LintTagger.fs

```diff
@@ -0,0 +1,105 @@
+            let! ast = checkResults.GetUntypedAst()
+            let res = 
+                Lint.lintParsedFile 
+                    { FinishEarly = None
+                      Configuration = None //Configuration.defaultConfiguration
+                      ReceivedWarning = None }
+                    { Ast = ast
+                      Source = source
+                      TypeCheckResults = checkResults.GetCheckResults() }
+                    doc.FullName
+
+            return
+                match res with
+                | LintResult.Success warnings ->
+                    warnings 
+                    |> Seq.distinctBy (fun warn -> warn.Range, warn.Info)
+                    |> Seq.choose (fun warn ->
+//                        let r = warn.Range
+//                        let endCol =
+//                            if r.StartLine = r.EndLine then
+//                                max r.EndColumn (r.StartColumn + 2)
+//                            else r.StartColumn + 2
+//                        let r' =    
+//                            Range.mkRange "" 
+//                                (Range.mkPos r.StartLine r.StartColumn)
+//                                (Range.mkPos r.StartLine endCol)
+
+                        fromFSharpRange view.TextBuffer.CurrentSnapshot warn.Range
+                        |> Option.map (fun span -> warn, span))
+                    |> Seq.toList
+                | LintResult.Failure _ -> []
+        }
+        |> Async.map (fun spans -> 
+            let spans = 
+                match spans with 
+                | Some x -> x
+                | None -> []
+            wordSpans <- spans
+            let span = SnapshotSpan(buffer.CurrentSnapshot, 0, buffer.CurrentSnapshot.Length)
+            tagsChanged.Trigger(self, SnapshotSpanEventArgs span))
+        |> Async.StartInThreadPoolSafe
+
+    let docEventListener = new DocumentEventListener ([ViewChange.bufferEvent textDocument.TextBuffer], 200us, updateAtCaretPosition)
+
+    let getTags (spans: NormalizedSnapshotSpanCollection): ITagSpan<LintTag> list = 
+        [
+            match wordSpans with
+            | [] -> ()
+            | _ ->
+                let currentSnapshot = spans.[0].Snapshot
+                let wordSpans = 
+                    if currentSnapshot = (snd wordSpans.[0]).Snapshot then
+                        wordSpans
+                    else
+                        // If the requested snapshot isn't the same as the one our words are on, translate our spans
+                        // to the expected snapshot
+                        wordSpans |> List.map (fun (warn, span) -> warn, span.TranslateTo(currentSnapshot, SpanTrackingMode.EdgeExclusive))
+                
+                for warn, span in wordSpans do
+                    if spans.OverlapsWith span then
+                        yield upcast TagSpan<LintTag>(span, LintTag(warn.Info))
```

Comment:
I think you can manipulate `spans` here if you would like to display a sub range of what FSharpLint gives.